provision/run-provisioners.sh

```diff
@@ -0,0 +1,307 @@
+
+function pre_hook() {
+  # provison-pre.sh
+  #
+  # acts as a pre-hook to our default provisioning script. Anything that
+  # should run before the shell commands laid out in provision.sh (or your provision-custom.sh
+  # file) should go in this script. If it does not exist, no extra provisioning will run.
+  if [[ -f "/srv/provision/provision-pre.sh" ]]; then
+    provisioner_begin "pre"
+    if [[ $? -ne "0" ]]; then
+      return 0
+    fi
+    bash /srv/provision/provision-pre.sh
+    PROVISION_SUCCESS=$?
+    provisioner_end "pre"
+  fi
+}
+
+function post_hook() {
+  # provision-post.sh
+  #
+  # acts as a post-hook to the default provisioning. Anything that should
+  # run after the shell commands laid out in provision.sh or provision-custom.sh should be
+  # put into this file. This provides a good opportunity to install additional packages
+  # without having to replace the entire default provisioning script.
+  if [[ -f "/srv/provision/provision-post.sh" ]]; then
+    provisioner_begin "post"
+    if [[ $? -ne "0" ]]; then
+      return 0
+    fi
+    bash /srv/provision/provision-post.sh
+    PROVISION_SUCCESS=$?
+    provisioner_end "post"
+  fi
+}
+
+function dashboard() {
+  provisioner_begin
+  if [[ $? -ne "0" ]]; then
+    return 0
+  fi
+  local dashboard_repo=$(get_config_value "dashboard.repo" "https://github.com/Varying-Vagrant-Vagrants/dashboard.git")
+  local dashboard_branch=$(get_config_value "dashboard.branch" "master")
+
+  bash /srv/provision/provision-dashboard.sh "${dashboard_repo}" "${dashboard_branch}"
+  PROVISION_SUCCESS=$?
+  provisioner_end
+}
+
+function utility_sources() {
+  local name=()
+  local repo=()
+  local branch=()
+
+  local key="utility-sources"
+
+  local utilities=$(get_config_type "${key}")
+  if [[ "${utilities}" == "struct" ]]; then
+    utilities=($(get_config_keys "${key}"))
+  else
+    utilities=$(get_config_value "${key}")
+    if [[ ! -z "${utilities}" ]]; then
+      vvv_error "Malformed ${key} config"
+    fi
+    utilities=()
+  fi
+
+  containsElement "core" "${utilities}"
+  if [[ $? -ne 0 ]]; then
+    name+=("core")
+    repo+=("https://github.com/Varying-Vagrant-Vagrants/vvv-utilities.git")
+    branch+=("master")
+  fi
+
+  local utility
+  for utility in "${utilities[@]}"; do
+    type=$(get_config_type "${key}.${utility}")
+    name+=(${utility})
+    if [[ "${type}" == "str" ]]; then
+      repo+=($(get_config_value "${key}.${utility}"))
+      branch+=(master)
+    else
+      repo+=($(get_config_value "${key}.${utility}.repo"))
+      branch+=($(get_config_value "${key}.${utility}.branch" "master"))
+    fi
+  done
+
+  local i
+  for i in ${!name[@]}; do
+    provisioner_begin "utility-source-${name[$i]}"
+    if [[ $? -ne "0" ]]; then
+      return 0
+    fi
+    bash /srv/provision/provision-utility-source.sh "${name[$i]}" "${repo[$i]}" "${branch[$i]}"
+    PROVISION_SUCCESS=$?
+    provisioner_end "utility-source-${name[$i]}"
+  done
+}
+
+function utility() {
```

Comment:
We should rename this to `provision_utilities`, and then take the inner loop and replace it with a `provision_utility` function so that we can provision a single utility if we really want to

Reply:
Done